lib/song.rb

```diff
@@ -0,0 +1,60 @@
+class Song
+  extend Concerns::Findable
+
+  @@all = []
+
+  def initialize(name, artist = nil, genre = nil)
+    @name = name
+    @artist = artist
+    if artist
+      artist.add_song(self) unless artist.songs.include?(self)
```

Comment:
Your **add_song** method already checks whether a song is included before adding. No need for the *unless*.